components/workspace-rollout-job/pkg/analysis/analyzer.go

```diff
@@ -21,3 +25,24 @@ type Analyzer interface {
 	// repeatedly to determine whether to move forward on the rollout or not.
 	MoveForward(ctx context.Context, clusterName string) (Decision, error)
 }
+
+func CheckPrometheusReachable(ctx context.Context, prometheusURL string) error {
+	if prometheusURL == "" {
+		return nil
+	}
```

Comment:
What about moving this function to `WorkspaceKeyMetricsAnalyzer`, e.g. `WorkspaceKeyMetricsAnalyzer.CheckPrometheusReachable()`, and then use the `api` client of the analyzer to check if it's reachable. You'd only need the `v1Client.Query(ctx, "up", time.Now())` part then, deduplicates the construction of the API clients